### PR TITLE
Fix delete user

### PR DESCRIPTION
**Pull request by mvn-haphan-dn**
## Description
A few sentences describing the overall goals of the pull request's commits.
## Reference Requirements
- Task ID: TI-60

## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

-

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)
```
explain delete from `ratings` where `ratings`.`user_id` = 2 and `ratings`.`user_id` is not null\G
*************************** 1. row ***************************
           id: 1
  select_type: DELETE
        table: ratings
   partitions: NULL
         type: range
possible_keys: ratings_user_id_foreign
          key: ratings_user_id_foreign
      key_len: 4
          ref: const
         rows: 2
     filtered: 100.00
        Extra: Using where
```
```
explain delete from `favorites` where `favorites`.`user_id` = 2 and `favorites`.`user_id` is not null\G
*************************** 1. row ***************************
           id: 1
  select_type: DELETE
        table: favorites
   partitions: NULL
         type: range
possible_keys: favorites_user_id_foreign
          key: favorites_user_id_foreign
      key_len: 4
          ref: const
         rows: 1
     filtered: 100.00
        Extra: Using where
```
```
explain update `posts` set `deleted_at` = "2018-05-29 01:23:18", `updated_at` = "2018-05-29 01:23:18" where `posts`.`user_id` = 2 and `posts`.`user_id` is not null\G
*************************** 1. row ***************************
           id: 1
  select_type: UPDATE
        table: posts
   partitions: NULL
         type: range
possible_keys: posts_user_id_foreign
          key: posts_user_id_foreign
      key_len: 4
          ref: const
         rows: 3
     filtered: 100.00
        Extra: Using where
```
```
explain update `users` set `deleted_at` = "2018-05-29 01:23:18", `updated_at` = "2018-05-29 01:23:18" where `id` = 2\G
*************************** 1. row ***************************
           id: 1
  select_type: UPDATE
        table: users
   partitions: NULL
         type: range
possible_keys: PRIMARY
          key: PRIMARY
      key_len: 4
          ref: const
         rows: 1
     filtered: 100.00
        Extra: Using where
```
```
explain select * from `borrowes` where `user_id` = 2 and `borrowes`.`deleted_at` is null\G
*************************** 1. row ***************************
           id: 1
  select_type: SIMPLE
        table: borrowes
   partitions: NULL
         type: ref
possible_keys: borrowes_user_id_foreign
          key: borrowes_user_id_foreign
      key_len: 4
          ref: const
         rows: 2
     filtered: 10.00
        Extra: Using where
```
```
 explain delete from `borrow_details` where `borrow_id` = 3\G
*************************** 1. row ***************************
           id: 1
  select_type: DELETE
        table: borrow_details
   partitions: NULL
         type: range
possible_keys: borrow_details_borrow_id_foreign
          key: borrow_details_borrow_id_foreign
      key_len: 4
          ref: const
         rows: 1
     filtered: 100.00
        Extra: Using where
```
```
explain delete from `borrow_details` where `borrow_id` = 9\G
*************************** 1. row ***************************
           id: 1
  select_type: DELETE
        table: borrow_details
   partitions: NULL
         type: range
possible_keys: borrow_details_borrow_id_foreign
          key: borrow_details_borrow_id_foreign
      key_len: 4
          ref: const
         rows: 1
     filtered: 100.00
        Extra: Using where
```
```
explain update `borrowes` set `deleted_at` = "2018-05-29 01:36:17", `updated_at` = "2018-05-29 01:36:17" where `borrowes`.`user_id` = 2 and `borrowes`.`user_id` is not null\G 
*************************** 1. row ***************************
           id: 1
  select_type: UPDATE
        table: borrowes
   partitions: NULL
         type: range
possible_keys: borrowes_user_id_foreign
          key: borrowes_user_id_foreign
      key_len: 4
          ref: const
         rows: 2
     filtered: 100.00
        Extra: Using where
```
- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

File: app/Http/Controllers/Admin/UserController.php [hunk 1]
```diff
@@ -129,8 +136,29 @@ public function store(CreateUserRequest $request)
     */
     public function destroy(User $user)
     {
-        $user->delete();
-        Session::flash('message', trans('user.messages.delete_success'));
+        //dd($user);
```
Comment (at-haytran) [inline app/Http/Controllers/Admin/UserController.php]: xóa dd

File: app/Http/Controllers/Admin/UserController.php [hunk 2]
```diff
@@ -129,8 +136,29 @@ public function store(CreateUserRequest $request)
...
+        if ($user->role == 0) {
```
Comment (at-haytran) [inline app/Http/Controllers/Admin/UserController.php]: User::ROLE_USER

File: app/Http/Controllers/Admin/UserController.php [hunk 3]
```diff
@@ -129,8 +136,29 @@ public function store(CreateUserRequest $request)
...
+            DB::beginTransaction();
+            try {
+                $user->ratings()->delete();
+                $user->favorites()->delete();
+                $user->posts()->delete();
+                $user->delete();
+                $borrowes = Borrow::where('user_id', $user->id)->get();
+                
+                if ($borrowes->count() > 0) {
```
Comment (at-haytran) [inline app/Http/Controllers/Admin/UserController.php]: chỗ này ($borrowes->count()) được rồi

File: app/Http/Controllers/Admin/UserController.php [hunk 4]
```diff
@@ -129,8 +136,29 @@ public function store(CreateUserRequest $request)
...
+        //dd($user);
+        if ($user->role == 0) {
+            DB::beginTransaction();
+            try {
+                $user->ratings()->delete();
+                $user->favorites()->delete();
+                $user->posts()->delete();
+                $user->delete();
+                $borrowes = Borrow::where('user_id', $user->id)->get();
+
```
Comment (at-haytran) [inline app/Http/Controllers/Admin/UserController.php]: xóa dòng trống

**Comment by at-haytran** [comment]
Check comment please

**Comment by at-haytran** [comment]
LGTM

File: resources/lang/en/user.php [hunk 5]
```diff
@@ -13,10 +13,15 @@
         'title' => 'Manage users',
         'create' => 'Create users',
         'update' => 'Update users',
-        'messages' => [
```
Comment (at-vietpham) [inline resources/lang/en/user.php]: Nên sửa thành theo cấu trúc 
```
├── messages
│   ├── success
│   ├── fail
│   ├── confirm
└───────
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by at-haytran** [comment]
LGTM

**Comment by at-vietpham** [comment]
LGTM 🚦

**Pull request merged**